include/poketch_data.h

```diff
@@ -49,245 +54,149 @@ enum PoketchScreenColor {
     POKETCH_SCREEN_COLOR_MAX,
 };
 
-enum {
-    POKETCH_MAPMARKER_COUNT = 6,
-    POKETCH_REGISTRY_SIZE = 32
-};
 
 /**
  * @brief All of the Poketch's internally tracked data including settings, registered apps, and the data for some apps (eg. pedometer, alarm clock).
  */
 typedef struct PoketchData {
-    u8 unk_00_0 : 1;
-    u8 pedometerEnabled : 1;    //!< Whether or not the pedometer is registered and will take step count updates.
-    u8 unk_00_2 : 1;
-    u8 screenColor : 3;         //!< Screen palette color (see PoketchScreenColor enum above)
-    u8 unk_00_6 : 2;            //!< unused; bitfield padding
+    u8 poketchEnabled : 1;
+    u8 pedometerEnabled : 1;        //!< Whether or not the pedometer is registered and will take step count updates.
+    u8 dotArtModifiedByPlayer : 1;  //!< Whether or not the dot art data has ever been modified by the user. This flag can never be set to FALSE.
+    u8 screenColor : 3;             //!< Screen palette color (see PoketchScreenColor enum above)
+    // u8 padding : 2;
 
     s8 appCount;                                //!< Number of currently registered apps
     s8 appIndex;                                //!< Currently selected app
     u8 appRegistry[POKETCH_REGISTRY_SIZE];      //!< Registration status of all apps. Indices 0-24 correspond to the App IDs in poketch_data.h. Indices 25-31 are unused.
     
-    u32 pedometer;         //!< Step counter
+    u32 stepCount;
 
-    u16 alarmSet : 1;      //!< Whether or not the alarm is currently enabled.
-    u16 alarmHour : 5;     //!< Current Hour setting on the alarm clock.
-    u16 alarmMinute : 6;   //!< Current Minute setting on the alarm clock.
-    u16 unk_28_12 : 4;     //!< unused; bitfield padding
+    u16 alarmSet : 1;
+    u16 alarmHour : 5;
+    u16 alarmMinute : 6;
+    // u16 padding : 4;
 
-    u8 unk_2A[120];
+    u8 dotArtData[POKETCH_DOTART_SIZE_BYTES];   //!< All pixel information for the Dot Art app
 
     u32 calendarMarkBitmap;     //!< Bitmap for every day in the calendar month and whether it's been marked
-    u8 calendarMonth;      //!< Current calendar month
+    u8 calendarMonth;
 
     /**
     * @brief XY coordinates of a single Map Marker in the Marking Map app
     */
     struct {
-        u8 x;                                       //!< X coordinate of map marker
-        u8 y;                                       //!< Y coordinate of map marker
-    } markMapPositions[POKETCH_MAPMARKER_COUNT];    //!< Map markers
+        u8 x;
+        u8 y;
+    } markMapPositions[POKETCH_MAPMARKER_COUNT];
 
+    /**
+    * @brief Relevant data for Pokemon History app
+    */
     struct {
-        u16 unk_00;
-        u16 unk_02;
-        u32 unk_04;
-    } unk_B8[12];
+        u16 species;
+        u16 icon;
+        u32 form;
+    } pokemonHistory[POKETCH_POKEMONHISTORY_MAX];   //!< The pokemon to display in the Pokemon History app
```

Comment:
Maybe it makes sense to name this in a way that clarifies that the operations on it treat it like a queue with bounded size.